src/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -558,7 +575,19 @@ function discoverController($scope, config, courier, $route, $window, Notifier,
   // TODO: On array fields, negating does not negate the combination, rather all terms
   $scope.filterQuery = function (field, values, operation) {
     $scope.indexPattern.popularizeField(field, 1);
-    filterManager.add(field, values, operation, $state.index);
+
```

Comment:
Any reason we don't want to have this stuff in the `filterManager` itself instead of having it semi-duplicated through discover, visualize, and dashboard?